src/main/java/org/checkerframework/specimin/UnusedImportRemoverVisitor.java

```diff
@@ -0,0 +1,273 @@
+      // Check for both cases: java.lang.Math.PI and java.lang.Math.*
+      usedImports.add(declaringType + "." + asField.getName());
+      usedImports.add(declaringType + ".*");
+    } else if (resolved.isEnumConstant()) {
+      ResolvedEnumConstantDeclaration asEnumConstant = resolved.asEnumConstant();
+      String declaringType = JavaParserUtil.erase(asEnumConstant.getType().describe());
+      // com.example.Enum, com.example.Enum.VALUE, com.example.Enum.*
+      usedImports.add(declaringType);
+      usedImports.add(declaringType + "." + asEnumConstant.getName());
+      usedImports.add(declaringType + ".*");
+    }
+    return super.visit(expr, arg);
+  }
+
+  @Override
+  public Visitable visit(MethodCallExpr expr, Void arg) {
+    ResolvedMethodDeclaration resolved;
+    try {
+      resolved = expr.resolve();
+    } catch (UnsupportedOperationException ex) {
+      // Lambdas can raise an UnsupportedOperationException
+      return super.visit(expr, arg);
+    } catch (UnsolvedSymbolException | IllegalStateException ex) {
+      unsolvedMembers.add(expr.getNameAsString());
+      return super.visit(expr, arg);
+    }
+
+    if (resolved.isStatic()) {
+      // If it has a scope, the parent class is imported
+      if (expr.hasScope()) {
+        handleScopeExpression(expr.getScope().get());
+      } else {
+        // Handle statically imported methods
+        // import static java.lang.Math.sqrt;
+        // sqrt(1);
+        // Check for both cases: java.lang.Math.sqrt and java.lang.Math.*
+        usedImports.add(JavaParserUtil.erase(resolved.getQualifiedName()));
+        usedImports.add(JavaParserUtil.erase(resolved.declaringType().getQualifiedName()) + ".*");
+      }
+    }
+
+    return super.visit(expr, arg);
+  }
+
+  @Override
+  public Visitable visit(MarkerAnnotationExpr anno, Void arg) {
+    handleAnnotation(anno);
+
+    return super.visit(anno, arg);
+  }
+
+  @Override
+  public Visitable visit(NormalAnnotationExpr anno, Void arg) {
+    handleAnnotation(anno);
+
+    return super.visit(anno, arg);
+  }
+
+  @Override
+  public Visitable visit(SingleMemberAnnotationExpr anno, Void arg) {
+    handleAnnotation(anno);
+
+    return super.visit(anno, arg);
+  }
+
+  /**
+   * Helper method to handle the scope type in a FieldAccessExpr or MethodCallExpr.
+   *
+   * @param scope The scope as an Expression
+   */
+  private void handleScopeExpression(Expression scope) {
+    // Workaround for a JavaParser bug: see UnsolvedSymbolVisitor#visit(ClassOrInterfaceType)
+    if (!JavaParserUtil.isCapital(scope.toString())) {
+      return;
+    }
+
+    String fullyQualified = JavaParserUtil.erase(scope.calculateResolvedType().describe());
+
+    if (!fullyQualified.contains(".")) {
+      // If there is no ., it is not a class (i.e. this.values.length)
+      return;
+    }
+
+    String wildcard = fullyQualified.substring(0, fullyQualified.lastIndexOf('.')) + ".*";
```

Comment:
I would extract this call to substring into a helper method, since it's repeated elsewhere (e.g., at line 130 above and just below at line 267).